### PR TITLE
Refactor/shared model compile time query responses

### PR DESCRIPTION
**Pull request by Akvinikym**
### Description of the Change

Moved definitions of methods, constructors and some local variables of queries responses in interfaces and backend protobuf into separate .cpp files from headers. It reduced build time of Iroha (target irohad) - the process was launched via ```make -j3 irohad```:
Before: ```user 1344.03s system 103.16s```. Total 1 447,19s
After: ```user 1290.79s, system 100.76s```. Total 1 391,55s

### Benefits

The build time was reduced by 4.00%

### Possible Drawbacks 

More files to maintain

### CODE REVIEW

File: shared_model/interfaces/query_responses/account_asset_response.hpp [hunk 1]
```diff
@@ -36,25 +33,9 @@ namespace shared_model {
...
-       * Stringify the data.
-       * @return string representation of data.
-       */
-      std::string toString() const override {
-        return detail::PrettyStringBuilder()
-            .init("AccountAssetResponse")
-            .append(accountAsset().toString())
-            .finalize();
-      }
+      std::string toString() const override;
```
Comment (x3medima17) [inline shared_model/interfaces/query_responses/account_asset_response.hpp]: Why did you remove docs? 📜

Comment (Akvinikym) [reply shared_model/interfaces/query_responses/account_asset_response.hpp]: I consulted with a couple of people and concluded that those comments were much too obvious and thus unnecessary. Do you think they should take place?

Comment (muratovv) [reply shared_model/interfaces/query_responses/account_asset_response.hpp]: Yeah! This documentation isn't looked useful. Just remove it.

Documentation of overloaded methods should contain useful details of implementation and/or changes in semantic.

File: shared_model/interfaces/query_responses/impl/account_asset_response.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/query_responses/account_asset_response.hpp"
+#include "utils/visitor_apply_for_all.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/impl/account_asset_response.cpp]: ToStringVisitor is not used here.

File: shared_model/interfaces/query_responses/impl/account_detail_response.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/query_responses/account_detail_response.hpp"
+#include "utils/visitor_apply_for_all.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/impl/account_detail_response.cpp]: ToStringVisitor is not used here.

File: shared_model/backend/protobuf/query_responses/proto_transaction_response.hpp [hunk 4]
```diff
@@ -33,32 +33,21 @@ namespace shared_model {
...
 
-      TransactionsResponse(TransactionsResponse &&o)
-          : TransactionsResponse(std::move(o.proto_)) {}
+      TransactionsResponse(TransactionsResponse &&o);
 
-      interface::types::TransactionsCollectionType transactions()
-          const override {
-        return *transactions_;
-      }
+      interface::types::TransactionsCollectionType transactions() const override;
```
Comment (lebdron) [inline shared_model/backend/protobuf/query_responses/proto_transaction_response.hpp]: Please apply clang format. Probably this is required in other files also.
